feature_engine/creation/combine_with_reference_feature.py

```diff
@@ -286,6 +293,9 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         if self.new_variables_names:
             X.columns = original_col_names + self.new_variables_names
 
+        if self.drop_original:
+            X.drop(columns=self.variables_, inplace=True)
```

Comment:
here we need to drop `self.variables_co_combine + self.reference_variables`.

I don't think this transformer has the attribute `variables_`, does it?